dbg.h

```diff
@@ -453,6 +457,16 @@ inline bool pretty_print(std::ostream& stream, const std::tuple<>&) {
   return true;
 }
 
+template <>
+inline bool pretty_print(std::ostream& stream, const time&) {
+  const auto ms =  std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count() % 1000000;
```

Comment:
`ms` is slightly confusing, as it usually describes milliseconds. I'll rename it to `us`.